config/src/main/java/com/typesafe/config/ConfigMemorySize.java

```diff
@@ -26,16 +29,36 @@ private ConfigMemorySize(long bytes) {
      * @param bytes a number of bytes
      * @return an instance representing the number of bytes
      */
-    public static ConfigMemorySize ofBytes(long bytes) {
+    public static ConfigMemorySize ofBytes(BigInteger bytes) {
         return new ConfigMemorySize(bytes);
     }
 
+    /**
+     * Constructs a ConfigMemorySize representing the given
+     * number of bytes.
+     * @param bytes a number of bytes
+     * @return an instance representing the number of bytes
+     */
+    public static ConfigMemorySize ofBytes(long bytes) {
+        return new ConfigMemorySize(BigInteger.valueOf(bytes));
+    }
+
     /**
      * Gets the size in bytes.
+     *
+     * @deprecated use {@link #getBytes()} to handle to bytes conversion of huge memory units.
```

Comment:
I'm a little tempted to not deprecate this, and name `getBytes` something more descriptive like `toBytesBigInteger`, since BigInteger would be perhaps a rare use-case.

Reply:
sure, fixed. The new method is named as `toBytesBigInteger()`, removed the deprecation tag from the original `toBytes()` method. Thank you.